vm/src/obj/objdict.rs

```diff
@@ -205,14 +205,31 @@ impl PyDictRef {
 
     #[cfg_attr(feature = "flame-it", flame("PyDictRef"))]
     fn inner_getitem(self, key: PyObjectRef, vm: &VirtualMachine) -> PyResult {
-        if let Some(value) = self.entries.borrow().get(vm, &key)? {
-            return Ok(value);
+        if let Some(value) = self.inner_getitem_option(&key, vm)? {
+            Ok(value)
+        } else {
+            Err(vm.new_key_error(key.clone()))
+        }
+    }
+
+    /// Return an optional inner item, or an error (can be key error as well)
+    fn inner_getitem_option<K: DictKey + IntoPyObject + Clone>(
+        &self,
+        key: &K,
```

Comment:
Here's an idea: implement `DictKey for Cow<str>`, since we only sometimes have to get an owned version of the `DictKey`.

Reply:
Not necessarily in this PR, but just something that crossed my mind.